x-pack/plugins/ml/public/application/datavisualizer/data_comparison/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,255 @@
+    () =>
+      dataViewEditorServiceFactory({
+        initialValues: {
+          name: '',
+          type: INDEX_PATTERN_TYPE.DEFAULT,
+          indexPattern: initialProductionIndexPattern,
+        },
+        requireTimestampField: true,
+      }),
+    [dataViewEditorServiceFactory, initialProductionIndexPattern]
+  );
+
+  const combinedTimeFieldOptions$: Observable<Array<EuiComboBoxOptionOption<string>>> =
+    useMemo(() => {
+      return combineLatest([
+        referenceDataViewEditorService.timestampFieldOptions$,
+        productionDataViewEditorService.timestampFieldOptions$,
+      ]).pipe(
+        map(([referenceTimeFieldOptions, productionTimeFieldOptions]) => {
+          return intersectionBy<TimestampOption, TimestampOption>(
+            referenceTimeFieldOptions,
+            productionTimeFieldOptions,
+            (d) => d.fieldName
+          ).map(({ display, fieldName }) => ({
+            label: display,
+            value: fieldName,
+          }));
+        })
+      );
+    }, [referenceDataViewEditorService, productionDataViewEditorService]);
+
+  const combinedTimeFieldOptions = useObservable(combinedTimeFieldOptions$, []);
+
+  const [referenceIndexPattern, setReferenceIndexPattern] = useState<string>(
+    initialReferenceIndexPattern ?? ''
+  );
+  const [productionIndexPattern, setProductionIndexPattern] = useState<string>(
+    initialProductionIndexPattern ?? ''
+  );
+
+  const navigateToPath = useNavigateToPath();
+
+  const createDataViewAndRedirectToDataComparisonPage = async () => {
+    // Create adhoc data view
+    const indicesName = `${referenceIndexPattern},${productionIndexPattern}`;
+
+    const matchingDataViews = await dataViews.find(indicesName);
+    const timeFieldName =
+      Array.isArray(timeField) && timeField.length > 0 ? timeField[0].value : undefined;
+
+    let dataView;
+    if (
+      Array.isArray(matchingDataViews) &&
+      matchingDataViews.length === 1 &&
+      matchingDataViews[0].timeFieldName === timeFieldName
+    ) {
+      dataView = matchingDataViews[0];
+    } else {
+      if (canEditDataView) {
+        dataView = await dataViews.createAndSave({
+          title: indicesName,
+          name: indicesName,
+          timeFieldName,
+        });
+      } else {
+        dataView = await dataViews.create({
+          title: indicesName,
+          name: indicesName,
+          timeFieldName,
+        });
+      }
+    }
+    const dataViewId = dataView.id;
+    const url = await locator.getUrl({
+      page: ML_PAGES.DATA_COMPARISON,
+      pageState: {
+        index: dataViewId,
+        reference: referenceIndexPattern,
+        production: productionIndexPattern,
+        timeFieldName,
+      },
+    });
+
+    await navigateToPath(url);
+  };
+
+  return (
+    <EuiFlexGroup direction="column">
+      <EuiFlexGrid className="fieldEditor__flyoutPanels" gutterSize="xl" columns={2}>
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
```

Comment:
If you reselect reference / production index patterns for which a data view already exists, you get this error in the console. Looks like we should be able to proceed, using the data view which was created previously:

<img width="1207" alt="image" src="https://github.com/elastic/kibana/assets/7405507/2d4fc33a-9eaa-4e33-97ec-9cf323f25a65">